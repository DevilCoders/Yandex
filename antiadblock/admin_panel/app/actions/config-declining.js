export const OPEN_CONFIG_DECLINING = 'OPEN_CONFIG_DECLINING';
export const CLOSE_CONFIG_DECLINING = 'CLOSE_CONFIG_DECLINING';

export function openConfigDeclining(serviceId, labelId, id) {
    return {
        type: OPEN_CONFIG_DECLINING,
        serviceId,
        labelId,
        id
    };
}

export function closeConfigDeclining(id) {
    return {
        type: CLOSE_CONFIG_DECLINING,
        id
    };
}

