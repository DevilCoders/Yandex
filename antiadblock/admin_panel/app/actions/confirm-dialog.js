export const OPEN_CONFIRM_DIALOG = 'OPEN_CONFIRM_DIALOG';
export const CLOSE_CONFIRM_DIALOG = 'CLOSE_CONFIRM_DIALOG';

export function openConfirmDialog(message, callback) {
    return {
        type: OPEN_CONFIRM_DIALOG,
        message,
        callback
    };
}

export function closeConfirmDialog() {
    return {
        type: CLOSE_CONFIRM_DIALOG
    };
}
