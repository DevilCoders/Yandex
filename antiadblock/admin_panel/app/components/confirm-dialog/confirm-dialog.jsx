import React from 'react';
import PropTypes from 'prop-types';

import i18n from 'app/lib/i18n';

import Bem from 'app/components/bem/bem';
import Button from 'lego-on-react/src/components/button/button.react';
import Modal from 'lego-on-react/src/components/modal/modal.react';

export default class ConfirmDialog extends React.Component {
    constructor(props) {
        super();

        this.state = {
            visible: props.visible
        };

        this.onApply = this.onApply.bind(this);
        this.onCancel = this.onCancel.bind(this);
    }

    componentWillReceiveProps(newProps) {
        this.setState({
            visible: newProps.visible
        });
    }

    onApply() {
        this.setState({
            visible: false
        });
        this.props.callback(true);
    }

    onCancel() {
        this.setState({
            visible: false
        });
        this.props.callback(false);
    }

    render() {
        return (
            <Modal visible={this.state.visible}>
                <Bem
                    key='title'
                    block='modal'
                    elem='title'>
                    {this.props.message}
                </Bem>
                <Bem
                    key='actions'
                    block='modal'
                    elem='actions'>
                    <Button
                        theme='action'
                        view='default'
                        tone='grey'
                        size='s'
                        mix={{
                            block: 'modal',
                            elem: 'action'
                        }}
                        onClick={this.onApply}>
                        {i18n('common', 'yes')}
                    </Button>
                    <Button
                        theme='normal'
                        view='default'
                        tone='grey'
                        size='s'
                        mix={{
                            block: 'modal',
                            elem: 'action'
                        }}
                        onClick={this.onCancel}>
                        {i18n('common', 'no')}
                    </Button>
                </Bem>
            </Modal>
        );
    }
}

ConfirmDialog.propTypes = {
    callback: PropTypes.func,
    message: PropTypes.string,
    visible: PropTypes.bool
};
