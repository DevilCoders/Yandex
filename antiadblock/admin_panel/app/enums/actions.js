export const ACTIONS = {
    // Временно скрыт мониторинг на сервисах, т.к. он пока не работает
    // GRAPHS: 'graphs',
    CONFIGS: 'configs',
    AUDIT: 'audit',
    INFO: 'info',
    UTILS: 'utils',
    // ANTIADB-2636
    // HEALTH: 'health',
    RULES: 'rules',
    SBS_RESULT_CHECKS: 'sbs-result-checks'
};

export const SECRET_ACTIONS = {
    SCREENSHOTS: 'screenshots-checks',
    PROFILES: 'sbs-profiles',
    LOGS: 'logs'
};
