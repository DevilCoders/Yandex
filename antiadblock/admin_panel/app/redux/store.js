import {createStore, compose, applyMiddleware} from 'redux';
import reducer from 'app/reducers/index';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';

let composeEnhancers = compose;

export function configureStore(state = {}) {
    if (typeof window !== 'undefined' && window.__REDUX_DEVTOOLS_EXTENSION__) {
        composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__;
    }
    if (process.env.NODE_ENV === 'local') {
        return createStore(reducer, state, composeEnhancers(applyMiddleware(thunkMiddleware, createLogger())));
    }
    return createStore(reducer, state, composeEnhancers(applyMiddleware(thunkMiddleware)));
}
