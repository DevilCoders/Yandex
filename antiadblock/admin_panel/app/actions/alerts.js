import servicesApi from 'app/api/services';
import {setErrors, clearErrors} from 'app/actions/errors';
import {getServices, getErrors} from 'app/reducers/index';
import {getById} from 'app/reducers/errors';
import {timeIsOver} from 'app/lib/errors-logic';

export const START_ALERTS_LOADING = 'START_ALERTS_LOADING';
export const END_ALERTS_LOADING = 'END_ALERTS_LOADING';

export function fetchAlerts() {
    return (dispatch, getState) => {
        dispatch(startAlertsLoading());

        return servicesApi.fetchAlerts().then(alerts => {
            const errors = getById(getErrors(getState()), 'dashboard');
            if (errors && errors.length) {
                dispatch(clearErrors('dashboard'));
            }

            return dispatch(endAlertsLoading(alerts.items));
        }, error => {
            const lastUpdateTime = getServices(getState()).lastUpdateTimeAlerts;

            if (timeIsOver(lastUpdateTime)) {
                return dispatch(setErrors('dashboard', [error.message]));
            }
        });
    };
}

export function startAlertsLoading() {
    return {
        type: START_ALERTS_LOADING
    };
}

export function endAlertsLoading(items) {
    return {
        type: END_ALERTS_LOADING,
        items
    };
}
