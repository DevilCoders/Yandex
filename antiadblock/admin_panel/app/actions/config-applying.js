import configApi from 'app/api/config';
import {setGlobalErrors} from 'app/actions/errors';

export const OPEN_CONFIG_APPLYING = 'OPEN_CONFIG_APPLYING';
export const CLOSE_CONFIG_APPLYING = 'CLOSE_CONFIG_APPLYING';
export const START_ACTIVE_CONFIG_LOADING = 'START_ACTIVE_CONFIG_LOADING';
export const END_ACTIVE_CONFIG_LOADING = 'END_ACTIVE_CONFIG_LOADING';

export function openConfigApplying(serviceId, id, configData, target) {
    return {
        type: OPEN_CONFIG_APPLYING,
        serviceId,
        id,
        configData,
        target
    };
}

export function closeConfigApplying(id) {
    return {
        type: CLOSE_CONFIG_APPLYING,
        id
    };
}

export function fetchCurrentConfig(labelId, target) {
    return dispatch => {
        dispatch(startActiveConfigLoading(labelId, target));

        return configApi.fetchCurrentConfig(labelId, target).then(config => {
            return dispatch(endActiveConfigLoading(labelId, target, config));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startActiveConfigLoading(labelId, target) {
    return {
        type: START_ACTIVE_CONFIG_LOADING,
        labelId,
        target
    };
}

export function endActiveConfigLoading(labelId, target, config) {
    return {
        type: END_ACTIVE_CONFIG_LOADING,
        labelId,
        target,
        config
    };
}
