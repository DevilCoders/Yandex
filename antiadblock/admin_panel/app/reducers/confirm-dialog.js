import * as confirmDialogActions from 'app/actions/confirm-dialog';

export default function(state, action) {
    const {type} = action;
    const initialState = {
        visible: false,
        message: null,
        callback: null
    };

    state = state || initialState;

    switch (type) {
        case confirmDialogActions.OPEN_CONFIRM_DIALOG: {
            return {
                visible: true,
                message: action.message,
                callback: action.callback
            };
        }
        case confirmDialogActions.CLOSE_CONFIRM_DIALOG: {
            return initialState;
        }
        default: {
            return state;
        }
    }
}
