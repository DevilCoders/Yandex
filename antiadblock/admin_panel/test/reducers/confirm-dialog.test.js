import reducer from 'app/reducers/confirm-dialog';
import * as confirmDialogActionFixtures from '../fixtures/actions/confirm-dialog';
import {CONFIRM_DIALOG_EMPTY} from '../fixtures/store';

describe('confirm-dialog reducer', () => {
    const initialState = CONFIRM_DIALOG_EMPTY;

    it('should return the initial state', () => {
        expect(reducer(undefined, {})).toMatchSnapshot();
    });

    it('should handle OPEN_CONFIRM_DIALOG', () => {
        expect(
            reducer(initialState, confirmDialogActionFixtures.OPEN_CONFIRM_DIALOG)
        ).toMatchSnapshot();
    });

    it('should handle CLOSE_CONFIRM_DIALOG', () => {
        const afterOpen = reducer(initialState, confirmDialogActionFixtures.OPEN_CONFIRM_DIALOG);

        expect(
            reducer(afterOpen, confirmDialogActionFixtures.CLOSE_CONFIRM_DIALOG)
        ).toMatchSnapshot();
    });
});
