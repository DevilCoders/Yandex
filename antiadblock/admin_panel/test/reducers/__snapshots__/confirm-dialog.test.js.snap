// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`confirm-dialog reducer should handle CLOSE_CONFIRM_DIALOG 1`] = `
Object {
  "callback": null,
  "message": null,
  "visible": false,
}
`;

exports[`confirm-dialog reducer should handle OPEN_CONFIRM_DIALOG 1`] = `
Object {
  "callback": "test function",
  "message": "test message",
  "visible": true,
}
`;

exports[`confirm-dialog reducer should return the initial state 1`] = `
Object {
  "callback": null,
  "message": null,
  "visible": false,
}
`;
